app/clients/sms/__init__.py

```diff
@@ -23,6 +23,7 @@ class SmsClient(Client):
     def init_app(self, current_app, statsd_client):
         self.current_app = current_app
         self.statsd_client = statsd_client
+        self.from_number = self.current_app.config.get('FROM_NUMBER')
```

Comment:
is this ever used now? relatively surprised that this is decided at the sms client level rather than earlier on.

looking here[^1] i expect that `sender` is always set and we can make the arg non-optional and remove the member variable entirely.


[^1]: https://github.com/alphagov/notifications-api/blob/master/app/delivery/send_to_providers.py#L77

Reply:
Good spot. We can certainly make it a required arg: https://github.com/alphagov/notifications-api/pull/3493/commits/6cbaa1bd4af2e75628aae45c17c688d4f982eed0.

The ternary is harder: although the argument is always provided, the `sender` may still be `None`. We persist notifications in quite a few places and ways so it's quite hard to tell. I know:

- All services must have a default SMS sender, so I'd never _expect_ `None`.
- None of the notifications in our Production DB have null `reply_to_text`.
- The original code was added [a very long time ago](https://github.com/alphagov/notifications-api/commit/c29dd237027dcbe6bd2073aaa774d0f90039d75b#diff-5517993ae3247630cf07bd936b84e1382ba7c764e9377815a91e29efeebdea6bR74) so might be redundant.

I'm cautious about making this change as part of the PR, partly because removing it will be more work and partly because I'm not totally confident in just dropping it e.g. if the assumption breaks in future.

How about:

- A separate PR to look at that (after this one).
- Some more thinking about how to enforce its presence.

Reply:
Sounds good! Good sleuthing around the impact. I suspect that it's never None any more but could have been in the early days. Agree that it best belongs in another PR

As a compromise, how do we feel about adding a log line if sender==None that we can then check for in a couple of weeks?

Reply:
Sure, that's a good idea - still possible of course but should be very niche.

I've added a log in https://github.com/alphagov/notifications-api/pull/3493/commits/0980e093bafafd799fdd94c709a1a9ca981d1f09 and a task on the card to revisit.